### DISCUSSION TITLE
Allure.link and parameter not working

### DISCUSSION THREAD

**Discussion by royalidea** [allure-framework/allure2#2668] [Bug Reports] [answered]
I have a java project using junit5 and cucumber. I'm trying to add links and parameters to the test case but it's not working. The data is not added.

This does nothing:
```
Allure.link("Link", "Test")
Allure.parameter("Parameter", "TestParameter")
```

I also tried something like this without success:
```
        Allure.getLifecycle().updateTestCase(result -> {
            result.setParameters(parameters);
        });
```

while Allure.addAttachment() is working. Am i missing something?

**Comment by baev** [comment]
Links, parameters, and labels can only be added to the test, while steps and attachments can be added to both test and test fixtures (setup and teardown methods).

**Reply by royalidea** [reply]
> Links, parameters, and labels can only be added to the test, while steps and attachments can be added to both test and test fixtures (setup and teardown methods).

Ok but also when i use this in my cucumber step it's not working.

```
    @Given("I am logged in")
    public void iAmLoggedIn() {
    Allure.link("Link", "Test")
    Allure.parameter("Parameter", "TestParameter")
    }
```

**Reply by baev** [reply, marked as answer]
What version of `allure-cucumber7jvm` are you using? The issue has been fixed in recent releases. I just checked with' allure-cucumber7jvm@2.29.0,` and everything works as expected.

**Reply by royalidea** [reply]
I was on 2.28.0. It's working with 2.29.0
